tests/test_pattern.py

```diff
@@ -193,5 +202,107 @@ def swap(circuit, a, b):
+        for node in localpattern.nodes.values():
+            np.testing.assert_equal(node.seq, [])
+
+    def test_get_graph(self):
+        nqubits = 5
+        depth = 4
+        pairs = [(i, np.mod(i + 1, nqubits)) for i in range(nqubits)]
+        circuit = rc.generate_gate(nqubits, depth, pairs)
+        pattern = circuit.transpile()
+        nodes_ref, edges_ref = pattern.get_graph()
+
+        localpattern = pattern.get_local_pattern()
+        nodes, edges = localpattern.get_graph()
+
+        # nodes check
+        nodes_check1 = True
+        nodes_check2 = True
+        for node in nodes:
+            if node not in nodes_ref:
+                nodes_check1 = False
+        for node in nodes_ref:
+            if node not in nodes:
+                nodes_check2 = False
+        np.testing.assert_equal(nodes_check1, True)
+        np.testing.assert_equal(nodes_check2, True)
+
+        # edges check
+        edges_check1 = True
+        edges_check2 = True
+        for edge in edges:
+            edge_match = False
+            for edge_ref in edges_ref:
+                edge_match |= assert_equal_edge(edge, edge_ref)
+            if not edge_match:
+                edges_check1 = False
+        for edge in edges_ref:
+            edge_match = False
+            for edge_ref in edges:
+                edge_match |= assert_equal_edge(edge, edge_ref)
+            if not edge_match:
+                edges_check2 = False
+        np.testing.assert_equal(edges_check1, True)
+        np.testing.assert_equal(edges_check2, True)
+
+    def test_standardize(self):
+        nqubits = 5
+        depth = 4
+        pairs = [(i, np.mod(i + 1, nqubits)) for i in range(nqubits)]
+        circuit = rc.generate_gate(nqubits, depth, pairs)
+        pattern = circuit.transpile()
+        localpattern = pattern.get_local_pattern()
+        localpattern.standardize()
+        pattern = localpattern.get_pattern()
+        np.testing.assert_equal(pattern.is_standard(), True)
+        pattern.minimize_space()
+        state_p = pattern.simulate_pattern()
+        state_ref = circuit.simulate_statevector()
+        np.testing.assert_almost_equal(np.abs(np.dot(state_p.flatten().conjugate(), state_ref.flatten())), 1)
+
+    def test_shift_signals(self):
+        nqubits = 5
+        depth = 4
+        pairs = [(i, np.mod(i + 1, nqubits)) for i in range(nqubits)]
+        circuit = rc.generate_gate(nqubits, depth, pairs)
+        pattern = circuit.transpile()
+        localpattern = pattern.get_local_pattern()
+        localpattern.standardize()
+        localpattern.shift_signals()
+        pattern = localpattern.get_pattern()
+        np.testing.assert_equal(pattern.is_standard(), True)
+        pattern.minimize_space()
+        state_p = pattern.simulate_pattern()
+        state_ref = circuit.simulate_statevector()
+        np.testing.assert_almost_equal(np.abs(np.dot(state_p.flatten().conjugate(), state_ref.flatten())), 1)
+
```

Comment:
what happens if someone tries normal standardization first, and then local standardization by mistake? or vice versa? or if mixed with signalshifting, e.g. standardize -> signalshift -> standardize? it's edge case but might be worth considering (e.g. throwing warning if we know they cause problem)